Svc/ComQueue/ComQueue.cpp

```diff
@@ -0,0 +1,308 @@
+    }
+    // Allocate a single chunk of memory from the memory allocator. Memory recover is neither needed nor used.
+    bool recoverable = false;
+    this->m_allocation = this->m_allocator->allocate(this->m_allocationId, totalAllocation, recoverable);
+
+    // Each of the backing queue objects must be supplied memory to store the queued messages. These data regions are
+    // sub-portions of the total allocated data. This memory is passed out by looping through each queue in prioritized
+    // order and passing out the memory to each queue's setup method.
+    FwSizeType allocationOffset = 0;
+    for (FwIndexType i = 0; i < TOTAL_PORT_COUNT; i++) {
+        // Get current queue's allocation size and safety check the values
+        FwSizeType allocationSize = this->m_prioritizedList[i].depth * this->m_prioritizedList[i].msgSize;
+        FW_ASSERT(this->m_prioritizedList[i].index < static_cast<FwIndexType>(FW_NUM_ARRAY_ELEMENTS(this->m_queues)), this->m_prioritizedList[i].index);
+        FW_ASSERT((allocationSize + allocationOffset) <= totalAllocation, allocationSize, allocationOffset,
+                  totalAllocation);
+
+        // Setup queue's memory allocation, depth, and message size. Setup is skipped for a depth 0 queue
+        if (allocationSize > 0) {
+            this->m_queues[this->m_prioritizedList[i].index].setup(reinterpret_cast<U8*>(this->m_allocation) + allocationOffset,
+                                                       allocationSize, this->m_prioritizedList[i].depth,
+                                                             this->m_prioritizedList[i].msgSize);
+        }
+        allocationOffset += allocationSize;
+    }
+    // Safety check that all memory was used as expected
+    FW_ASSERT(allocationOffset == totalAllocation, allocationOffset, totalAllocation);
+}
+// ----------------------------------------------------------------------
+// Handler implementations for user-defined typed input ports
+// ----------------------------------------------------------------------
+
+void ComQueue::comQueueIn_handler(const NATIVE_INT_TYPE portNum, Fw::ComBuffer& data, U32 context) {
+    // Ensure that the port number of comQueueIn is consistent with the expectation
+    FW_ASSERT(portNum >= 0 && portNum < COM_PORT_COUNT, portNum);
+    this->enqueue(portNum, QueueType::COM_QUEUE, reinterpret_cast<const U8*>(&data), sizeof(Fw::ComBuffer));
+}
+
+void ComQueue::buffQueueIn_handler(const NATIVE_INT_TYPE portNum, Fw::Buffer& fwBuffer) {
+    const NATIVE_INT_TYPE queueNum = portNum + COM_PORT_COUNT;
+    // Ensure that the port number of buffQueueIn is consistent with the expectation
+    FW_ASSERT(portNum >= 0 && portNum < BUFFER_PORT_COUNT, portNum);
+    FW_ASSERT(queueNum < TOTAL_PORT_COUNT);
+    this->enqueue(queueNum, QueueType::BUFFER_QUEUE, reinterpret_cast<const U8*>(&fwBuffer), sizeof(Fw::Buffer));
+}
+
+void ComQueue::comStatusIn_handler(const NATIVE_INT_TYPE portNum, Fw::Success& condition) {
```

Comment:
This implementation of `comStatusIn_handler` does not agree with what the SDD says. See https://github.com/LeStarch/fprime/blob/update/com-queue/Svc/ComQueue/docs/sdd.md#463-comstatusin.